python/res/simulator/simulation_context.py

```diff
@@ -19,43 +15,34 @@ def __init__(self, ert, sim_fs, mask, itr , verbose=False):
         job_queue = ert.get_queue_config().create_job_queue()
         job_queue.set_max_job_duration(max_runtime)
         self._queue_manager = JobQueueManager(job_queue)
-        self._queue_manager.startQueue( mask.count( ), verbose=verbose)
+
         self._run_args = {}
         """ :type: dict[int, RunArg] """
 
-        self._thread_pool = CThreadPool(8)
-        self._thread_pool.addTaskFunction("submitJob", RES_LIB, "enkf_main_isubmit_job__")
-
         subst_list = self._ert.getDataKW( )
         path_fmt = self._ert.getModelConfig().getRunpathFormat()
         jobname_fmt = self._ert.getModelConfig().getJobnameFormat()
-        self._run_context = ErtRunContext( EnkfRunType.ENSEMBLE_EXPERIMENT, sim_fs, None, mask, path_fmt, jobname_fmt, subst_list, itr)
-        self._ert.initRun(self._run_context)
-
-
-    def __len__(self):
-        return self._mask.count()
 
+        self._run_context = ErtRunContext( EnkfRunType.ENSEMBLE_EXPERIMENT, sim_fs, None, mask, path_fmt, jobname_fmt, subst_list, itr)
+        for run_arg in self._run_context:
+            if run_arg is not None:
+                self._run_args[run_arg.iens] = run_arg
```

Comment:
Why are we storing both the run_context and run_args. Ownership of the run_args objects is not very clearly seen here.

Reply:
That's for convenience purposes. Agree, this can go away.